cli/src/psp-utils/createRustVerifyingKey.ts

```diff
@@ -105,28 +106,42 @@ async function getProofInputsFromSymFile(artifactPath: string) {
 
 function createStringRsIdlAccountStruct(
   preparedInputs: PropertiesObject[],
-  circuitName: string
+  circuitName: string,
+  prefix: string = "ZK",
+  isInstructionData: boolean = false
 ) {
   function camelToSnakeCase(str: string) {
     return str.replace(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`);
   }
 
-  function buildRustType(dimension: number, size: number[]): string {
+  function buildRustType(
+    dimension: number,
+    size: number[],
+    isInstructionData: boolean = false
+  ): string {
+    // the base type is a 254 bit bigint thus a 32 byte array
+    if (dimension === 0 && isInstructionData) {
+      return "[u8; 32]";
+    }
     if (dimension === 0) {
       return "u8";
     }
 
-    let rustType = buildRustType(dimension - 1, size.slice(1));
+    let rustType = buildRustType(
+      dimension - 1,
+      size.slice(1),
+      isInstructionData
+    );
     return `[${rustType};${size[0]}]`;
   }
 
   ///  MAIN FUNCTION START
   // parse the inputs output into a rust struct as a program account
-  let structDefinition = `\n#[account]\npub struct ZK${circuitName}Inputs {\n`;
+  let structDefinition = `\n#[account]\npub struct ${prefix}${circuitName} {\n`;
 
   preparedInputs.forEach((input) => {
     const { inputName, dimension, size } = input;
-    const rustType = buildRustType(dimension, size);
+    const rustType = buildRustType(dimension, size, isInstructionData);
     const inputName_snake = camelToSnakeCase(inputName);
     structDefinition += `    ${inputName_snake}: ${rustType},\n`;
```

Comment:
are the empty spaces before ${ planned? 
```suggestion
    structDefinition += `${inputName_snake}: ${rustType},\n`;
```